src/components/search-location.ts

```diff
@@ -0,0 +1,104 @@
+import { LitElement, html, unsafeCSS } from "lit";
+import { customElement, query, state } from "lit/decorators.js";
+import { Control } from "ol/control";
+import { useStore } from "../composable/store";
+import style from '../styles/search.css?inline';
+
+import * as olProj from 'ol/proj';
+import { unsafeSVG } from "lit/directives/unsafe-svg.js";
+import SVGCreator from "../utils/svg-creator";
+import { cache } from "lit/directives/cache.js";
+
+@customElement('search-location')
+class SearchLocation extends LitElement {
+  @query('#search')
+  public inputElement!: HTMLInputElement;
+  @query('#results')
+  public results!: HTMLUListElement;
+
+  @state() _hasSearch: boolean = false;
+
+  static styles = [unsafeCSS(style)];
+
+  firstUpdated() {
+    this.inputElement.oninput = () => {
+        if (this.inputElement.value.length > 1) {
+            this._hasSearch = true;
+            const options = useStore().getOptions();
+            let url = `${options.search.requestWithoutCustomValue}&searchText=${this.inputElement.value}`
+            if (options.search.bboxRestiction !== '') url += `&bbox=${options.search.bboxRestiction}`;
+            fetch(url).then((result) => result.json()).then((json) => {
+                if (json && json.results.length > 0) {
+                    const maxDisplayedLocation = json.results.length > 5 ? 5 : json.results.length;
+                    this.results.innerHTML = '';
+                    for (let i = 0; i < maxDisplayedLocation; i++){
+               
+                        const item = document.createElement('li')
+                        let label = '';
+
+                        item.addEventListener('click', () => {
+                            this.inputElement.value = label;
+                            useStore().getMap().getView().setCenter(olProj.fromLonLat([json.results[i].attrs.lon, json.results[i].attrs.lat], 'EPSG:2056'))
+                            this.results.innerHTML = '';
+                        })
+
+                        if (i === 0) item.classList.add('first-li')
+                        if (i === maxDisplayedLocation - 1) item.classList.add('last-li')
```

Comment:
why are you not using let templating? Would be probably easier to read since it would dissociate fetch logic and template.
`fetch` would update and internal object of results and in render you would `map` on result to display the list

Reply:
Done